espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/WebAtoms.kt

```diff
@@ -53,18 +53,22 @@ class WebAtoms : RequestHandler<WebAtomsParams, Void?> {
         // Iterate through methodsChain and call the atoms
         params.methodChain.forEach { method ->
             val atom = invokeStaticMethod(DriverAtoms::class.java, method.atom.name, *method.atom.args) as? Atom<*>
-                    ?: throw InvalidArgumentException("'${method.atom.name}' did not return a valid " +
-                            "'${Atom::class.qualifiedName}' object")
+                ?: throw InvalidArgumentException(
+                    "'${method.atom.name}' did not return a valid " +
+                        "'${Atom::class.qualifiedName}' object",
+                )
 
             AndroidLogger.info("Calling interaction '${method.name}' with the atom '${method.atom}'")
 
             val res = invokeInstanceMethod(webViewInteraction, method.name, atom) as? WebInteraction<*>
-                    ?: throw InvalidArgumentException("'${method.name}' does not return a valid " +
-                            "'${WebInteraction::class.qualifiedName}' object")
+                ?: throw InvalidArgumentException(
+                    "'${method.name}' does not return a valid " +
+                        "'${WebInteraction::class.qualifiedName}' object",
+                )
 
             webViewInteraction = res
         }
 
-        return null
+        return if (params.methodChain.size == 1) "success" else webViewInteraction.get().toString()
```

Comment:
I am not sure we always need to cast the returned result to string (it could also be null)
Also what is the meaning of `params.methodChain.size == 1`  condition?

Reply:
`params.methodChain.size == 1 ` for a scenario where I want to check if an element exists and not perform any action.

```bash
[HTTP] --> POST /wd/hub/session/cb9acb48-ccfb-4b17-82de-a7b8d4303c35/execute/sync
[HTTP] {"script":"mobile: webAtoms","args":[{"webviewElement":"4184be82-034d-4a0e-acc9-1af385c4e348","forceJavascriptEnabled":true,"methodChain":[{"name":"withElement","atom":{"name":"findElement","locator":{"using":"XPATH","value":"//*[@data-qa='safety-centre-item']"}}}]}]}
[debug] [EspressoDriver@a433 (cb9acb48)] Calling AppiumDriver.execute() with args: ["mobile: webAtoms",[{"webviewElement":"4184be82-034d-4a0e-acc9-1af385c4e348","forceJavascriptEnabled":true,"methodChain":[{"name":"withElement","atom":{"name":"findElement","locator":{"using":"XPATH","value":"//*[@data-qa='safety-centre-item']"}}}]}],"cb9acb48-ccfb-4b17-82de-a7b8d4303c35"]
[EspressoDriver@a433 (cb9acb48)] Executing native command 'mobile: webAtoms'
[debug] [EspressoDriver@a433 (cb9acb48)] Proxying [POST /appium/execute_mobile/web_atoms] to [POST http://127.0.0.1:52472/session/96b63585-c431-476e-9c37-555b9ebd4dcb/appium/execute_mobile/web_atoms] with body: {"webviewElement":"4184be82-034d-4a0e-acc9-1af385c4e348","forceJavascriptEnabled":true,"methodChain":[{"name":"withElement","atom":{"name":"findElement","locator":{"using":"XPATH","value":"//*[@data-qa='safety-centre-item']"}}}]}
[debug] [EspressoDriver@a433 (cb9acb48)] Got response with status 200: {"id":"a4c7bb04-f16e-41f1-a11d-0fef55e46a18","sessionId":"96b63585-c431-476e-9c37-555b9ebd4dcb","value":"success"}
```

Otherwise it throws below error
```bash
[EspressoDriver@7732 (d0303a0b)] Replacing sessionId f55230ec-2f73-44dc-a17c-8a12a999146a with d0303a0b-f8ea-497e-8a70-8e929c467ba9
[HTTP] <-- POST /wd/hub/session/d0303a0b-f8ea-497e-8a70-8e929c467ba9/element 200 71 ms - 232
[HTTP]
[HTTP] --> POST /wd/hub/session/d0303a0b-f8ea-497e-8a70-8e929c467ba9/execute/sync
[HTTP] {"script":"mobile: webAtoms","args":[{"webviewElement":"cd114239-cbe6-43c1-bbde-2376269c0336","forceJavascriptEnabled":true,"methodChain":[{"name":"withElement","atom":{"name":"findElement","locator":{"using":"XPATH","value":"//*[@data-qa='safety-centre-item']"}}}]}]}
[debug] [EspressoDriver@7732 (d0303a0b)] Calling AppiumDriver.execute() with args: ["mobile: webAtoms",[{"webviewElement":"cd114239-cbe6-43c1-bbde-2376269c0336","forceJavascriptEnabled":true,"methodChain":[{"name":"withElement","atom":{"name":"findElement","locator":{"using":"XPATH","value":"//*[@data-qa='safety-centre-item']"}}}]}],"d0303a0b-f8ea-497e-8a70-8e929c467ba9"]
[EspressoDriver@7732 (d0303a0b)] Executing native command 'mobile: webAtoms'
[debug] [EspressoDriver@7732 (d0303a0b)] Proxying [POST /appium/execute_mobile/web_atoms] to [POST http://127.0.0.1:58945/session/f55230ec-2f73-44dc-a17c-8a12a999146a/appium/execute_mobile/web_atoms] with body: {"webviewElement":"cd114239-cbe6-43c1-bbde-2376269c0336","forceJavascriptEnabled":true,"methodChain":[{"name":"withElement","atom":{"name":"findElement","locator":{"using":"XPATH","value":"//*[@data-qa='safety-centre-item']"}}}]}
[EspressoDriver@7732 (d0303a0b)] Got response with status 500: {"id":"30fe2de0-e3af-4b45-850b-7a3bda55564f","sessionId":"f55230ec-2f73-44dc-a17c-8a12a999146a","value":{"error":"unknown error","message":"java.lang.IllegalStateException: Perform or Check never called on this WebInteraction!","stacktrace":"java.lang.IllegalStateException: Perform or Check never called on this WebInteraction!\n\tat androidx.test.internal.util.Checks.checkState(Checks.java:107)\n\tat androidx.test.espresso.web.sugar.Web$WebInteraction.get(Web.java:324)\n\tat io.appium.espressoserver.lib.handlers.WebAtoms.handleInternal(WebAtoms.kt:72)\n\tat io.appium.espressoserver.lib.handlers.WebAtoms.handleInternal(WebAtoms.kt:32)\n\tat io.appium.espressoserver.lib.handlers.RequestHandler$DefaultImpls.handle(RequestHandler.kt:57)\n\tat io.appium.espressoserver.lib.handlers.WebAtoms.handle(WebAtoms.kt:32)\n\tat io.appium.espressoserver.lib.http.Router.route(Router.kt:231)\n\tat io.appium.espressoserver.lib.http.Server.serve(Server.kt:54)\n\tat fi.iki.elonen.NanoHTTPD$HTTPSession.execute(NanoHTTPD.java:94...
[debug] [W3C] Matched W3C error code 'unknown error' to UnknownError
[debug] [EspressoDriver@7732 (d0303a0b)] Encountered internal error running command: An unknown server-side error occurred while processing the command. Original error: java.lang.IllegalStateException: Perform or Check never called on this WebInteraction!
[debug] [EspressoDriver@7732 (d0303a0b)] java.lang.IllegalStateException: Perform or Check never called on this WebInteraction!
[debug] [EspressoDriver@7732 (d0303a0b)] 	at androidx.test.internal.util.Checks.checkState(Checks.java:107)
[debug] [EspressoDriver@7732 (d0303a0b)] 	at androidx.test.espresso.web.sugar.Web$WebInteraction.get(Web.java:324)
[debug] [EspressoDriver@7732 (d0303a0b)] 	at io.appium.espressoserver.lib.handlers.WebAtoms.handleInternal(WebAtoms.kt:72)
[debug] [EspressoDriver@7732 (d0303a0b)] 	at io.appium.espressoserver.lib.handlers.WebAtoms.handleInternal(WebAtoms.kt:32)
[debug] [EspressoDriver@7732 (d0303a0b)] 	at io.appium.espressoserver.lib.handlers.RequestHandler$DefaultImpls.handle(RequestHandler.kt:57)
[debug] [EspressoDriver@7732 (d0303a0b)] 	at io.appium.espressoserver.lib.handlers.WebAtoms.handle(WebAtoms.kt:32)
[debug] [EspressoDriver@7732 (d0303a0b)] 	at io.appium.espressoserver.lib.http.Router.route(Router.kt:231)
[debug] [EspressoDriver@7732 (d0303a0b)] 	at io.appium.espressoserver.lib.http.Server.serve(Server.kt:54)
[debug] [EspressoDriver@7732 (d0303a0b)] 	at fi.iki.elonen.NanoHTTPD$HTTPSession.execute(NanoHTTPD.java:945)
[debug] [EspressoDriver@7732 (d0303a0b)] 	at fi.iki.elonen.NanoHTTPD$ClientHandler.run(NanoHTTPD.java:192)
[debug] [EspressoDriver@7732 (d0303a0b)] 	at java.lang.Thread.run(Thread.java:1012)
[debug] [EspressoDriver@7732 (d0303a0b)]
[HTTP] <-- POST /wd/hub/session/d0303a0b-f8ea-497e-8a70-8e929c467ba9/execute/sync 500 131 ms - 1130
```

Reply:
Not calling `toSting()` works fine, will pull push changes in a few.